medic/views.py

```diff
@@ -1,3 +1,75 @@
-from django.shortcuts import render
 
-# Create your views here.
+from medic.models import Medic
+from medic.serializers import MedicSerializer
+from rest_framework import generics
+from rest_framework import permissions
+from medic.permissions import IsOwnerOrReadOnly
+from rest_framework.response import Response
+from rest_framework.exceptions import AuthenticationFailed
+from rest_framework.decorators import api_view
+from rest_framework.reverse import reverse
+import jwt, datetime
+
+
+    
+
+class MedicList(generics.ListCreateAPIView):
+    """
+
+    """
+    queryset = Medic.objects.all()
+    serializer_class = MedicSerializer
+    
+   
+    # TODO: implement and test authentication to uncomment the next line
+    def post(self, request):
+        email = request.data['email']
+        password = request.data['password']
+        
+                      
+        user = Medic.objects.filter(email=email).first()
+
+        if user is None:
```

Comment:
In this post view, you're checking if the user with email entered already exist, right, and if not, return user not found. On registration, the user isn't supposed to have their email already on the database :)